src/Aspire.Hosting.Azure.AppService/AzureAppServiceComputeResourceExtensions.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Azure.Provisioning.AppService;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for publishing compute resources as Azure App Service websites.
+/// </summary>
+public static class AzureAppServiceComputeResourceExtensions
+{
+    /// <summary>
+    /// Publishes the specified compute resource as an Azure App Service.
+    /// </summary>
+    /// <typeparam name="T">The type of the compute resource.</typeparam>
+    /// <param name="builder">The compute resource builder.</param>
+    /// <param name="configure">The configuration action for the App Service WebSite resource.</param>
+    /// <returns>The updated compute resource builder.</returns>
+    /// <remarks>
+    /// This method checks if the application is in publish mode. If it is, it adds the necessary infrastructure
+    /// for Azure App Service and applies the provided configuration action to the App Service WebSite resource.
+    /// <example>
+    /// <code>
+    /// builder.AddNpmApp("name", "image").PublishAsAzureAppServiceWebsite((infrastructure, app) =>
```

Comment:
This doesn't work yet, right? Should this be AddProject?

Reply:
Yep!